### ISSUE TITLE
At directory "tutorials/image/cifar10", running cifar10_train.py and cifar10_multi_gpu_train.py got similar trainning speed.

### ISSUE THREAD

**Issue by lipond** [tensorflow/models#1171]
## tutorials/image/cifar10

I use tf 0.11 in a gpu server with 2 K20m, equiped with CUDA-8.0 and cuDNN 5.1. I got similar trainning speed running cifar10_train.py and cifar10_multi_gpu_train.py, and did not see speeding using multi gpus. And using "watch -n 0.1 nvidia-smi" to monitor, gave the same conclusion.

## Running cifar10_train.py got following outputs:
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so locally
Filling queue with 20000 CIFAR images before starting to train. This will take a few minutes.
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 0 with properties:
name: Tesla K20m
major: 3 minor: 5 memoryClockRate (GHz) 0.7055
pciBusID 0000:02:00.0
Total memory: 4.63GiB
Free memory: 4.57GiB
W tensorflow/stream_executor/cuda/cuda_driver.cc:572] creating context when one is currently active; existing: 0x33c3ab0
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 1 with properties:
name: Tesla K20m
major: 3 minor: 5 memoryClockRate (GHz) 0.7055
pciBusID 0000:03:00.0
Total memory: 4.63GiB
Free memory: 4.57GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:972] DMA: 0 1
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 0:   Y Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 1:   Y Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K20m, pci bus id: 0000:02:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:1) -> (device: 1, name: Tesla K20m, pci bus id: 0000:03:00.0)
2017-03-15 14:36:15.050932: step 0, loss = 4.68 (6.8 examples/sec; 18.961 sec/batch)
2017-03-15 14:36:17.602299: step 10, loss = 4.63 (842.3 examples/sec; 0.152 sec/batch)
2017-03-15 14:36:19.130538: step 20, loss = 4.54 (791.0 examples/sec; 0.162 sec/batch)
2017-03-15 14:36:20.675717: step 30, loss = 4.39 (814.3 examples/sec; 0.157 sec/batch)
2017-03-15 14:36:22.219651: step 40, loss = 4.29 (811.0 examples/sec; 0.158 sec/batch)
2017-03-15 14:36:23.762940: step 50, loss = 4.32 (836.6 examples/sec; 0.153 sec/batch)
2017-03-15 14:36:25.289821: step 60, loss = 4.35 (817.5 examples/sec; 0.157 sec/batch)
2017-03-15 14:36:26.817753: step 70, loss = 4.29 (770.0 examples/sec; 0.166 sec/batch)
2017-03-15 14:36:28.344277: step 80, loss = 4.26 (806.8 examples/sec; 0.159 sec/batch)
2017-03-15 14:36:29.886840: step 90, loss = 4.12 (840.5 examples/sec; 0.152 sec/batch)
2017-03-15 14:36:31.402936: step 100, loss = 4.14 (912.0 examples/sec; 0.140 sec/batch)
2017-03-15 14:36:33.100713: step 110, loss = 4.53 (833.6 examples/sec; 0.154 sec/batch)
2017-03-15 14:36:34.612226: step 120, loss = 4.21 (882.0 examples/sec; 0.145 sec/batch)
2017-03-15 14:36:36.174803: step 130, loss = 3.88 (819.6 examples/sec; 0.156 sec/batch)
2017-03-15 14:36:37.697032: step 140, loss = 4.07 (845.8 examples/sec; 0.151 sec/batch)
2017-03-15 14:36:39.211350: step 150, loss = 4.03 (901.8 examples/sec; 0.142 sec/batch)
2017-03-15 14:36:40.725264: step 160, loss = 3.92 (794.8 examples/sec; 0.161 sec/batch)
2017-03-15 14:36:42.272620: step 170, loss = 3.79 (826.8 examples/sec; 0.155 sec/batch)
2017-03-15 14:36:43.822235: step 180, loss = 3.91 (836.8 examples/sec; 0.153 sec/batch)
2017-03-15 14:36:45.340961: step 190, loss = 3.87 (835.1 examples/sec; 0.153 sec/batch)
2017-03-15 14:36:46.853391: step 200, loss = 3.53 (866.9 examples/sec; 0.148 sec/batch)

## Running cifar10_multi_gpu_train.py got following outputs:
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcublas.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcudnn.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcufft.so locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcuda.so.1 locally
I tensorflow/stream_executor/dso_loader.cc:111] successfully opened CUDA library libcurand.so locally
Filling queue with 20000 CIFAR images before starting to train. This will take a few minutes.
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 0 with properties:
name: Tesla K20m
major: 3 minor: 5 memoryClockRate (GHz) 0.7055
pciBusID 0000:02:00.0
Total memory: 4.63GiB
Free memory: 4.57GiB
W tensorflow/stream_executor/cuda/cuda_driver.cc:572] creating context when one is currently active; existing: 0x2402260
I tensorflow/core/common_runtime/gpu/gpu_device.cc:951] Found device 1 with properties:
name: Tesla K20m
major: 3 minor: 5 memoryClockRate (GHz) 0.7055
pciBusID 0000:03:00.0
Total memory: 4.63GiB
Free memory: 4.57GiB
I tensorflow/core/common_runtime/gpu/gpu_device.cc:972] DMA: 0 1
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 0:   Y Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:982] 1:   Y Y
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:0) -> (device: 0, name: Tesla K20m, pci bus id: 0000:02:00.0)
I tensorflow/core/common_runtime/gpu/gpu_device.cc:1041] Creating TensorFlow device (/gpu:1) -> (device: 1, name: Tesla K20m, pci bus id: 0000:03:00.0)
2017-03-15 14:44:25.812755: step 0, loss = 4.68 (6.7 examples/sec; 19.150 sec/batch)
2017-03-15 14:44:28.450604: step 10, loss = 4.60 (831.3 examples/sec; 0.154 sec/batch)
2017-03-15 14:44:29.967788: step 20, loss = 4.41 (948.2 examples/sec; 0.135 sec/batch)
2017-03-15 14:44:31.525162: step 30, loss = 4.33 (804.9 examples/sec; 0.159 sec/batch)
2017-03-15 14:44:33.067339: step 40, loss = 4.37 (843.4 examples/sec; 0.152 sec/batch)
2017-03-15 14:44:34.612613: step 50, loss = 4.43 (817.6 examples/sec; 0.157 sec/batch)
2017-03-15 14:44:36.161372: step 60, loss = 4.15 (833.1 examples/sec; 0.154 sec/batch)
2017-03-15 14:44:37.709379: step 70, loss = 4.24 (877.5 examples/sec; 0.146 sec/batch)
2017-03-15 14:44:39.279416: step 80, loss = 4.15 (792.1 examples/sec; 0.162 sec/batch)
2017-03-15 14:44:40.837732: step 90, loss = 4.25 (833.8 examples/sec; 0.154 sec/batch)
2017-03-15 14:44:42.410801: step 100, loss = 4.17 (843.9 examples/sec; 0.152 sec/batch)
2017-03-15 14:44:44.112250: step 110, loss = 4.21 (796.1 examples/sec; 0.161 sec/batch)
2017-03-15 14:44:45.669297: step 120, loss = 4.02 (836.9 examples/sec; 0.153 sec/batch)
2017-03-15 14:44:47.212069: step 130, loss = 4.15 (1073.4 examples/sec; 0.119 sec/batch)
2017-03-15 14:44:48.731499: step 140, loss = 3.99 (838.7 examples/sec; 0.153 sec/batch)
2017-03-15 14:44:50.330916: step 150, loss = 4.16 (801.7 examples/sec; 0.160 sec/batch)
2017-03-15 14:44:51.877821: step 160, loss = 3.92 (777.1 examples/sec; 0.165 sec/batch)
2017-03-15 14:44:53.418757: step 170, loss = 3.84 (810.2 examples/sec; 0.158 sec/batch)
2017-03-15 14:44:54.983055: step 180, loss = 3.84 (826.7 examples/sec; 0.155 sec/batch)
2017-03-15 14:44:56.542567: step 190, loss = 3.74 (792.6 examples/sec; 0.161 sec/batch)
2017-03-15 14:44:58.094476: step 200, loss = 3.96 (799.5 examples/sec; 0.160 sec/batch)
2017-03-15 14:44:59.797948: step 210, loss = 3.97 (830.3 examples/sec; 0.154 sec/batch)
2017-03-15 14:45:01.326099: step 220, loss = 3.69 (786.5 examples/sec; 0.163 sec/batch)
2017-03-15 14:45:02.855803: step 230, loss = 4.05 (833.1 examples/sec; 0.154 sec/batch)
2017-03-15 14:45:04.401365: step 240, loss = 3.65 (754.8 examples/sec; 0.170 sec/batch)
2017-03-15 14:45:05.925520: step 250, loss = 3.59 (831.0 examples/sec; 0.154 sec/batch)
2017-03-15 14:45:07.501809: step 260, loss = 3.71 (822.4 examples/sec; 0.156 sec/batch)
2017-03-15 14:45:09.059610: step 270, loss = 3.54 (804.2 examples/sec; 0.159 sec/batch)
2017-03-15 14:45:10.611221: step 280, loss = 4.07 (851.8 examples/sec; 0.150 sec/batch)
2017-03-15 14:45:12.160260: step 290, loss = 3.40 (823.7 examples/sec; 0.155 sec/batch)
2017-03-15 14:45:13.735393: step 300, loss = 3.48 (805.0 examples/sec; 0.159 sec/batch)
2017-03-15 14:45:15.468795: step 310, loss = 3.58 (841.8 examples/sec; 0.152 sec/batch)
2017-03-15 14:45:17.023397: step 320, loss = 3.80 (787.5 examples/sec; 0.163 sec/batch)
2017-03-15 14:45:18.528321: step 330, loss = 3.53 (904.5 examples/sec; 0.142 sec/batch)
2017-03-15 14:45:20.080703: step 340, loss = 3.64 (870.8 examples/sec; 0.147 sec/batch)
2017-03-15 14:45:21.646686: step 350, loss = 3.35 (819.4 examples/sec; 0.156 sec/batch)
2017-03-15 14:45:23.184980: step 360, loss = 3.50 (864.7 examples/sec; 0.148 sec/batch)
2017-03-15 14:45:24.740313: step 370, loss = 3.37 (798.9 examples/sec; 0.160 sec/batch)

**Comment by lipond** [comment]
I figure out it: in line 59, the "num_gpus" needs to be set as 2, and the output:

2017-03-15 19:27:40.337271: step 1790, loss = 1.53 (1725.5 examples/sec; 0.074 sec/batch)
2017-03-15 19:27:41.805715: step 1800, loss = 1.75 (1686.7 examples/sec; 0.076 sec/batch)
2017-03-15 19:27:43.446241: step 1810, loss = 1.87 (1702.8 examples/sec; 0.075 sec/batch)
2017-03-15 19:27:44.892425: step 1820, loss = 1.73 (1786.9 examples/sec; 0.072 sec/batch)
2017-03-15 19:27:46.359524: step 1830, loss = 1.68 (1731.4 examples/sec; 0.074 sec/batch)
2017-03-15 19:27:47.816298: step 1840, loss = 1.65 (1757.5 examples/sec; 0.073 sec/batch)
2017-03-15 19:27:49.283323: step 1850, loss = 1.53 (1755.7 examples/sec; 0.073 sec/batch)
2017-03-15 19:27:50.744904: step 1860, loss = 1.49 (1877.0 examples/sec; 0.068 sec/batch)
2017-03-15 19:27:52.203909: step 1870, loss = 1.57 (1769.9 examples/sec; 0.072 sec/batch)
2017-03-15 19:27:53.660454: step 1880, loss = 1.68 (1749.3 examples/sec; 0.073 sec/batch)
2017-03-15 19:27:55.118941: step 1890, loss = 1.51 (1702.4 examples/sec; 0.075 sec/batch)
2017-03-15 19:27:56.565108: step 1900, loss = 1.74 (1757.1 examples/sec; 0.073 sec/batch)
2017-03-15 19:27:58.200960: step 1910, loss = 1.65 (1764.9 examples/sec; 0.073 sec/batch)
2017-03-15 19:27:59.666017: step 1920, loss = 1.58 (1714.4 examples/sec; 0.075 sec/batch)
2017-03-15 19:28:01.117179: step 1930, loss = 1.71 (1753.0 examples/sec; 0.073 sec/batch)
2017-03-15 19:28:02.572699: step 1940, loss = 1.86 (1776.4 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:04.029885: step 1950, loss = 1.64 (1779.0 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:05.487640: step 1960, loss = 1.37 (1723.8 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:06.941088: step 1970, loss = 1.51 (1772.5 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:08.407358: step 1980, loss = 1.71 (1730.1 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:09.867160: step 1990, loss = 1.49 (1723.4 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:11.322493: step 2000, loss = 1.58 (1715.3 examples/sec; 0.075 sec/batch)
2017-03-15 19:28:14.455045: step 2010, loss = 1.81 (1717.5 examples/sec; 0.075 sec/batch)
2017-03-15 19:28:15.909846: step 2020, loss = 1.59 (1800.3 examples/sec; 0.071 sec/batch)
2017-03-15 19:28:17.373296: step 2030, loss = 1.44 (1839.5 examples/sec; 0.070 sec/batch)
2017-03-15 19:28:18.835678: step 2040, loss = 1.35 (1716.0 examples/sec; 0.075 sec/batch)
2017-03-15 19:28:20.280054: step 2050, loss = 1.54 (1768.2 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:21.739694: step 2060, loss = 1.41 (1730.3 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:23.210179: step 2070, loss = 1.67 (1787.7 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:24.668458: step 2080, loss = 1.38 (1729.1 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:26.122845: step 2090, loss = 1.66 (1750.6 examples/sec; 0.073 sec/batch)
2017-03-15 19:28:27.581868: step 2100, loss = 1.26 (1781.9 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:29.219163: step 2110, loss = 1.64 (1831.1 examples/sec; 0.070 sec/batch)
2017-03-15 19:28:30.684090: step 2120, loss = 1.53 (1766.3 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:32.141261: step 2130, loss = 1.46 (1796.4 examples/sec; 0.071 sec/batch)
2017-03-15 19:28:33.593914: step 2140, loss = 1.30 (1769.6 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:35.065221: step 2150, loss = 1.44 (1827.0 examples/sec; 0.070 sec/batch)
2017-03-15 19:28:36.519679: step 2160, loss = 1.55 (1752.2 examples/sec; 0.073 sec/batch)
2017-03-15 19:28:37.987297: step 2170, loss = 1.43 (1734.5 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:39.458737: step 2180, loss = 1.33 (1705.9 examples/sec; 0.075 sec/batch)
2017-03-15 19:28:40.912454: step 2190, loss = 1.45 (1731.0 examples/sec; 0.074 sec/batch)
2017-03-15 19:28:42.361551: step 2200, loss = 1.51 (1768.6 examples/sec; 0.072 sec/batch)
2017-03-15 19:28:43.998409: step 2210, loss = 1.85 (1742.2 examples/sec; 0.073 sec/batch)
2017-03-15 19:28:45.462839: step 2220, loss = 1.49 (1772.9 examples/sec; 0.072 sec/batch)
